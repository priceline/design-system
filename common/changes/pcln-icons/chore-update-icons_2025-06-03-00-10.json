{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Update to styled-system v5 on pcln-icons, move to dev dependency and peer dependency. Use Node v20 for the build process.",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}