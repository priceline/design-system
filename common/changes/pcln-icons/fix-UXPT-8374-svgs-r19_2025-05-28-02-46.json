{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "replace defaultProps usage in svgr template file",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}