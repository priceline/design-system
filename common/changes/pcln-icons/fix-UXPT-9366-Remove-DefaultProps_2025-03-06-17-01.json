{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "remove defaultProps",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}