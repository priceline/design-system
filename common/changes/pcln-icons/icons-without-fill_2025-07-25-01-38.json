{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "UXPT-10383: rounded Material SVGs from filled to unfilled",
      "type": "major"
    }
  ],
  "packageName": "pcln-icons"
}