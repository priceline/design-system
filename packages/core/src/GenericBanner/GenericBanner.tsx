import React from 'react'
import styled from 'styled-components'
import { AlignItemsProps, FontSizeProps, JustifyContentProps, MarginProps, PaddingProps } from 'styled-system'
import { Banner } from '../Banner/Banner'
import { Box } from '../Box/Box'
import { Button } from '../Button/Button'
import { Flex } from '../Flex/Flex'
import { Link } from '../Link/Link'
import { Relative } from '../Relative/Relative'
import { ColorSchemeName } from '../theme/theme'
import { colorSchemeCustomForeground } from '../utils/utils'

const BannerWithRadius = styled(Banner)`
  cursor: ${(props) => (props.onClick ? 'pointer' : 'cursor')};

  ${colorSchemeCustomForeground}
`

const WrapperLink = styled.a`
  z-index: 1;
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`

const CustomLink = styled(Link)`
  display: inline-flex;
  z-index: 2;
`

const CustomButton = styled(Button)`
  z-index: 2;
  text-decoration: ${(props) => (props.buttonTextUnderline ? 'underline' : 'none')};
`

/**
 * @public
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type GenericBannerNode = React.ReactElement<any, string | React.JSXElementConstructor<any>>

/**
 * @public
 */
export type GenericBannerProps = AlignItemsProps &
  JustifyContentProps &
  FontSizeProps &
  PaddingProps &
  MarginProps & {
    buttonClick?: () => unknown
    buttonSize?: 'small' | 'medium' | 'large'
    buttonVariation?: 'fill' | 'outline' | 'link'
    ctaText?: GenericBannerNode
    heading?: GenericBannerNode
    iconLeft?: GenericBannerNode
    iconRight?: GenericBannerNode
    imageLeft?: GenericBannerNode
    linkVariation?: 'fill' | 'outline' | 'link'
    borderRadius?: string
    linkColor?: string
    text?: GenericBannerNode
    URLProps?: {
      href: string
      target?: string
    }
    color?: string
    colorScheme?: ColorSchemeName
  }

/**
 * @public
 */
export function GenericBanner({
  alignItems = 'center',
  borderRadius = '6px',
  buttonClick,
  buttonSize,
  buttonVariation = 'link',
  ctaText,
  fontSize = [0, null, null, 1],
  heading,
  iconLeft,
  iconRight,
  imageLeft,
  justifyContent = 'center',
  linkVariation = 'link',
  linkColor = 'primary',
  text,
  /* eslint-disable @typescript-eslint/naming-convention */
  URLProps,
  ...props
}: GenericBannerProps) {
  function onClick(e) {
    e.stopPropagation()
    buttonClick()
  }

  return (
    <Relative>
      {URLProps && <WrapperLink tabIndex={-1} aria-hidden='true' {...URLProps} />}
      <BannerWithRadius
        {...props}
        onClick={URLProps ? null : buttonClick}
        iconUsesColorScheme
        // we need to unset Banner's color prop if colorScheme is provided so it doesn't render its own icon
        color={!props.colorScheme ? props.color : null}
        borderRadius={borderRadius}
      >
        <Flex alignItems={alignItems} justifyContent={justifyContent}>
          {!!iconLeft && iconLeft}
          {!!imageLeft && imageLeft}
          {/*
            fontSize is not a prop Box supports
            <Box px={2} fontSize={fontSize}>
          */}
          <Box px={2}>
            {!!heading &&
              React.cloneElement(heading, {
                fontSize: heading.props.fontSize || fontSize,
              })}
            {!!text &&
              React.cloneElement(text, {
                fontSize,
              })}

            {URLProps && ctaText && (
              <CustomLink
                p={linkVariation === 'fill' || linkVariation === 'outline' ? 2 : null}
                color={linkColor}
                variation={linkVariation}
                fontSize={fontSize}
                {...URLProps}
              >
                {React.cloneElement(ctaText, {
                  fontSize,
                  zIndex: 2,
                })}
              </CustomLink>
            )}
            {buttonClick && typeof buttonClick === 'function' && ctaText && (
              <CustomButton
                size={buttonSize}
                buttonTextUnderline={buttonVariation === 'link'}
                variation={buttonVariation}
                onClick={onClick}
              >
                {React.cloneElement(ctaText, {
                  fontSize,
                })}
              </CustomButton>
            )}
          </Box>
          {!!iconRight && iconRight}
        </Flex>
      </BannerWithRadius>
    </Relative>
  )
}
