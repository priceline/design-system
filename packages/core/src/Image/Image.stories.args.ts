import { ArgTypes } from '@storybook/react/*'
import { borderRadii, colors, objectFits, objectPositions, rounded, shadows } from '../storybook/args'

export const argTypes: ArgTypes = {
  borderRadius: {
    control: 'select',
    name: 'borderRadius',
    options: borderRadii,
  },
  rounded: {
    control: 'select',
    options: rounded,
  },
  boxShadowSize: {
    control: 'select',
    options: shadows,
  },
  boxShadowColor: {
    control: 'select',
    options: colors,
    defaultValue: 'border',
    description: 'Palette family for box shadows',
    table: {
      type: {
        summary: 'Palette color to use for box-shadow - always uses "dark" shade',
      },
    },
  },
  height: {
    control: 'text',
  },
  width: {
    control: 'text',
  },
  objectFit: {
    control: 'select',
    options: objectFits,
  },
  objectPosition: {
    control: 'select',
    options: objectPositions,
  },
}
