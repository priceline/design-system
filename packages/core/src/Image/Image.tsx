import React, { type ComponentPropsWithoutRef } from 'react'
import styled from 'styled-components'
import {
  BorderRadiusProps,
  BoxShadowProps,
  HeightProps,
  MaxHeightProps,
  MaxWidthProps,
  MinHeightProps,
  MinWidthProps,
  SpaceProps,
  WidthProps,
  borderRadius,
  boxShadow,
  compose,
  height,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  space,
  width,
} from 'styled-system'
import {
  ObjectFit,
  ObjectPosition,
  borderRadiusAttrs,
  boxShadowAttrs,
  objectFit,
  objectPosition,
} from '../utils'

/**
 * @public
 */
export type ImageProps = BorderRadiusProps &
  BoxShadowProps &
  HeightProps &
  MaxHeightProps &
  MaxWidthProps &
  MinHeightProps &
  MinWidthProps &
  SpaceProps &
  WidthProps &
  Omit<ComponentPropsWithoutRef<'img'>, 'width' | 'height'> & {
    borderRadiusSize?: string
    rounded?: string
    boxShadowSize?: string
    boxShadowColor?: string
    objectFit?: ObjectFit
    objectPosition?: ObjectPosition
  }

/**
 * @public
 */
export const StyledImg: React.FC<ImageProps> = styled.img.attrs((props) => ({
  ...borderRadiusAttrs(props),
  ...boxShadowAttrs(props),
}))`
  display: block;
  ${objectFit};
  ${objectPosition};

  ${(props) =>
    compose(width, height, maxHeight, maxWidth, minHeight, minWidth, space, borderRadius, boxShadow)(props)}
`

/**
 * @public
 */
export function Image({ boxShadowColor = 'border', maxWidth = '100%', ...props }: ImageProps) {
  return <StyledImg boxShadowColor={boxShadowColor} maxWidth={maxWidth} {...props} />
}

Image.displayName = 'Image'
