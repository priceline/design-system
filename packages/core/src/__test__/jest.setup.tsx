import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
