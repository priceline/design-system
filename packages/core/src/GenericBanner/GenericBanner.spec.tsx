import { Emoticon, Warning } from 'pcln-icons'
import { fireEvent, render } from '../__test__/testing-library'
import { Image } from '../Image/Image'
import { Text } from '../Text/Text'
import { GenericBanner } from './GenericBanner'

const props = {
  p: 2,
  heading: (
    <Text.span fontWeight='bold' textColor='primary.base' mr={1}>
      COVID-19
    </Text.span>
  ),
  text: (
    <Text.span textColor='primary.base' mr={1}>
      Update: Your travel may be impacted. Please review this hotels important info.
    </Text.span>
  ),
  ctaText: (
    <Text.span textColor='primary.base' fontWeight='bold'>
      Read More
    </Text.span>
  ),
  imageLeft: <Image alt='ImageLeft' />,
  iconLeft: <Warning />,
  iconRight: <Emoticon />,
  color: 'caution.light',
}

const propsUpdatedHeading = {
  ...props,
  heading: (
    <Text.span fontWeight='bold' textColor='primary.base' mr={1} fontSize={4}>
      COVID-19
    </Text.span>
  ),
}

describe('GenericBanner', () => {
  test('renders', () => {
    const { asFragment } = render(<GenericBanner />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Renders as expected-onclick', () => {
    const buttonClick = jest.fn()
    const { getByText } = render(<GenericBanner {...props} buttonClick={buttonClick} />)
    const header = getByText('COVID-19')
    const text = getByText('Update: Your travel may be impacted. Please review this hotels important info.')
    expect(header).toBeTruthy()
    expect(text).toBeTruthy()
    const readMoreButton = getByText('Read More')
    expect(readMoreButton).toBeTruthy()
    fireEvent.click(readMoreButton)
    expect(buttonClick).toHaveBeenCalled()
    expect(readMoreButton).toHaveStyleRule('font-size', '12px')
    expect(header).toHaveStyleRule('font-size', '12px')
    expect(text).toHaveStyleRule('font-size', '12px')
  })
  test('Renders as expected- url', () => {
    const { getByText, getByRole } = render(
      <GenericBanner
        {...props}
        URLProps={{
          href: 'https://www.priceline.com',
          target: '_blank',
        }}
      />
    )
    const header = getByText('COVID-19')
    const text = getByText('Update: Your travel may be impacted. Please review this hotels important info.')
    expect(getByRole('img')).toBeTruthy()
    expect(header).toBeTruthy()
    expect(text).toBeTruthy()
    const readMoreButton = getByText('Read More')
    expect(readMoreButton).toBeTruthy()
    fireEvent.click(readMoreButton)
    expect(readMoreButton).toHaveStyleRule('font-size', '12px')
    expect(header).toHaveStyleRule('font-size', '12px')
    expect(text).toHaveStyleRule('font-size', '12px')
  })
  test('Renders as expected- with different heading size', () => {
    const { getByText, getByRole } = render(
      <GenericBanner
        {...propsUpdatedHeading}
        URLProps={{
          href: 'https://www.priceline.com',
          target: '_blank',
        }}
      />
    )
    const header = getByText('COVID-19')
    const text = getByText('Update: Your travel may be impacted. Please review this hotels important info.')
    expect(getByRole('img')).toBeTruthy()
    expect(header).toBeTruthy()
    expect(text).toBeTruthy()
    const readMoreButton = getByText('Read More')
    expect(readMoreButton).toBeTruthy()
    fireEvent.click(readMoreButton)
    expect(readMoreButton).toHaveStyleRule('font-size', '12px')
    expect(header).toHaveStyleRule('font-size', '24px')
    expect(text).toHaveStyleRule('font-size', '12px')
  })
})
