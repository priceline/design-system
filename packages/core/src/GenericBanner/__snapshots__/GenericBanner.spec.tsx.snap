// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericBanner renders 1`] = `
<DocumentFragment>
  .c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c6 {
  outline: none;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c8 {
  font-size: 14px;
}

.c2 {
  background-color: #0a0;
  color: #fff;
  border-radius: 6px;
  text-align: left;
}

.c7 {
  text-align: left;
  width: 100%;
}

.c10 {
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  cursor: cursor;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        color="text.lightest"
      >
        <div
          class="c4"
        >
          <svg
            aria-hidden="true"
            class="c5 c6"
            fill="currentcolor"
            focusable="false"
            height="24"
            role="img"
            size="24"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
            />
          </svg>
          <div
            class="c7"
            width="1"
          >
            <span
              class="c8"
              font-size="1"
            />
            <div
              class="c9"
            >
              <div
                class="c10"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
